scripts/ga_airtable_transfer.js

```diff
@@ -0,0 +1,149 @@
+  creds = require('../../sheetsjwt.json')
+} else {
+  creds = {
+    client_email: process.env.GOOGLE_SHEET_EMAIL,
+    private_key: process.env.GOOGLE_SHEET_KEY
+  }
+}
+
+function authenticateGoogle () {
+  return new Promise((resolve, reject) => {
+    doc.useServiceAccountAuth(creds, (err, res) => {
+      err ? reject(err) : resolve(res)
+    })
+  })
+}
+
+function docInfo () {
+  return new Promise((resolve, reject) => {
+    doc.getInfo((err, res) => {
+      err ? reject(err) : resolve(res.worksheets[1].rowCount)
+    })
+  })
+}
+
+async function copyDataTask (offset) {
+  try {
+    const getRows = () =>
+      new Promise((resolve, reject) => {
+        doc.getRows(2, { limit: 5, offset: offset }, (err, rows) => {
+          err ? reject(err) : resolve(rows)
+        })
+      })
+    const rows = await getRows()
+    let promises = rows.map((row) => airtableFetch(row))
+    await Promise.all(promises)
+  } catch (err) {
+    console.log(err)
+  }
+}
+
+async function airtableFetch (row) {
+  try {
+    const getRecord = () =>
+      new Promise((resolve, reject) => {
+        base('Archive').select({
+          filterByFormula: "({Name} = 'https://" + landingPage + "')",
+          view: 'Archive List'
+        }).firstPage((err, recs) => {
+          err ? reject(err) : resolve(recs)
+        })
+      })
+
+    const updateRecord = (recordId, row) =>
+      new Promise((resolve, reject) => {
+        base('Archive').update(recordId, rowValues(row, false), (err, record) => {
+          err ? reject(err) : resolve(record)
+        })
+      })
+
+    const createRecord = (row) =>
+      new Promise((resolve, reject) => {
+        base('Archive').create(rowValues(row, true), (err, record) => {
+          err ? reject(err) : resolve(record)
+        })
+      })
+
+    const landingPage = row.galandingpagepath
+    if (!landingPage) {
+      return
+    }
+    const records = await getRecord()
+
+    // Create or Update Airtable record
+    if (!records.length) {
+      const create = await createRecord(row)
+      return create
+    } else {
+      if (!(records && records[0].getId())) {
+        return
+      }
+      var recordId = records[0].getId()
+      const update = await updateRecord(recordId, row)
+      return update
+    }
+  } catch (err) {
+    console.log(err)
+  }
+}
+function rowValues (row) {
+  return {
+    gausers: parseInt(row.gausers),
+    gasessions: parseInt(row.gasessions),
+    gaavgSessionDuration: parseFloat(row.gaavgsessionduration),
+    gapageviewsPerSession: parseFloat(row.gapageviewspersession),
+    'Name': 'https://' + row.galandingpagepath
+  }
+}
+
+(async function () {
+  try {
+    // Google Auth
+    await authenticateGoogle()
+    console.log('Completed Authentication')
+
+    // Google Sheet Info
+    rowCount = await docInfo()
+    console.log('Total Rows: ' + rowCount)
+    var rowBuckets = Math.ceil(rowCount / 5)
+
+    // Copy Data Tasks
+    for (var i = 0; i <= rowBuckets; i++) {
```

Comment:
Might be nicer to use `forEach` here. e.g.

```javascript
rowBuckets.forEach(async (_, i) => await copyDataTask(i * 5))
```

Or, if reasonable, you could even do `copyDataTask` in parallel with `map` e.g.

```javascript
const copyTasks = rowBuckets.map((_, i) => copyDataTask(i * 5))
await Promise.all(copyTasks)
```